src/Components/Components/src/Rendering/RenderTreeBuilder.cs

```diff
@@ -581,6 +581,14 @@ public void AddComponentReferenceCapture(int sequence, Action<object> componentR
         /// <param name="sequence">An integer that represents the position of the instruction in the source code.</param>
         public void OpenRegion(int sequence)
         {
+            // We are entering a new scope, since we track the "duplicate attributes" per
+            // element/component we might need to clean them up now.
+            if (_hasSeenAddMultipleAttributes)
+            {
+                var indexOfLastElementOrComponent = _openElementIndices.Peek();
+                ProcessDuplicateAttributes(first: indexOfLastElementOrComponent + 1);
```

Comment:
I haven't fully wrap my head around this yet, but CloseComponent and CloseElement also do housekeeping and closeRegion doesn't seem to. Do we need to account for that?

https://github.com/aspnet/AspNetCore/pull/16640/files#diff-c09589b3f0826f7ed0084bef9e8816c2R532-R535

Reply:
I don’t think that’s necessary, since regions can’t have attributes of their own.

Reply:
Understood